src/Workspaces/Core/Portable/Options/IOption.cs

```diff
@@ -11,5 +12,7 @@ public interface IOption
         Type Type { get; }
         object DefaultValue { get; }
         bool IsPerLanguage { get; }
+
+        ImmutableArray<OptionSerialization> Serializations { get; }
```

Comment:
is OptionSerialization supposed to be just strongly typed data rather than code (which can have its own behavior)? asking since having behavior given to it will make serializing options harder.

one question, in current form, if I want to serialize all options to a stream, do you think it can be easily done?